freedomofpress/securedrop-client
# Add black and isort

# Description

Adds the black and isort tools to the development requirements, and adds Makefile targets for both checking and applying them.

Fixes #1100.

# Test Plan

- Check out the first commit on `add-black-isort` with `git checkout 86ee815`
- Run `make check-isort`. It should show the changes it would make.
- Run `make check-black`. It should show the changes it would make.
- Check out the last commit: `git checkout add-black-isort`
- Rerun `make check-isort` and `make check-black`. No changes should be indicated.
- Run `git blame securedrop_client/logic.py | head -60 | grep -c 361ba88`. There should be 22 lines attributed to that revision.
- Run: 
  - `git config blame.ignoreRevsFile .git-blame-ignore-revs`
  - `git config blame.markIgnoredLines true`
  - `git config blame.markUnblamableLines true`
- Run `git blame securedrop_client/logic.py | head -60 | grep 361ba88`. There should be three lines attributed to that revision, each prefixed with an asterisk. This indicates that those changes are only attributable to the ignored revision.



# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:


If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:


If these changes modify the database schema, you should include a database migration. Please check as applicable:
